### PR TITLE
Feature/first stage enrichment polygons

### PR DESCRIPTION
**Pull request by alejandrohall**
Closes https://github.com/CartoDB/data-observatory/issues/174
Closes https://github.com/CartoDB/data-observatory/issues/142
Closes https://github.com/CartoDB/data-observatory/issues/143

### CODE REVIEW

File: cartoframes/data/clients/bigquery_client.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        job = self.client.load_table_from_dataframe(dataframe, table_ref, job_config=job_config)
+        job.result()
+
+    def query(self, query, **kwargs):
+        response = self.client.query(query, **kwargs)
+
+        return response
+
+    def delete_table(self, tablename, project, dataset):
```
Comment (alrocar) [inline cartoframes/data/clients/bigquery_client.py]: We have not decided yet how to delete tables, but at the moment the role we'll use in BQ does not have delete permission.

That means when we integrate with the `token` API, we might have to disable deletion of tables.

Comment (oleurud) [reply cartoframes/data/clients/bigquery_client.py]: :+1:

File: cartoframes/data/enrichment/points_enrichment.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+              NULL AS population
+        FROM `carto-do-customers.{user_workspace}.{enrichment_table}` enrichment_table
+        JOIN `carto-do-customers.{user_workspace}.{enrichment_geo_table}` enrichment_geo_table
+          ON enrichment_table.geoid = enrichment_geo_table.geoid
+        JOIN `{working_project}.{working_dataset}.{data_table}` data_table
+          ON ST_Within(data_table.{data_geom_column}, enrichment_geo_table.geom)
+        {filters};
+    '''.format(enrichment_id=enrichment_id, variables=', '.join(variables),
+               enrichment_table=enrichment_table, enrichment_geo_table=enrichment_geo_table,
+               user_workspace=user_workspace.replace('-', '_'), working_project=working_project,
```
Comment (oleurud) [inline cartoframes/data/enrichment/points_enrichment.py]: A detail: I think `.replace('-', '_')` it is already done

File: cartoframes/data/enrichment/points_enrichment.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+# TODO: process column name in metadata, remove spaces and points
+
+
+def enrich_points(data, variables, data_geom_column='geometry', filters=dict(), credentials=None):
+
+    credentials = credentials or get_default_credentials()
+    bq_client = bigquery_client.BigQueryClient(credentials)
+
+    user_workspace = credentials.username.replace('-', '_')
```
Comment (oleurud) [inline cartoframes/data/enrichment/points_enrichment.py]: A comment: get the user dataset name, should it be a responsibility of credentials class? If yes, I will add it to https://github.com/CartoDB/data-observatory/issues/147 issue

A detail: should we call it dataset? avoiding using several names for the same thing

File: cartoframes/data/enrichment/points_enrichment.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+    data_copy = data_copy.merge(data_geometry_id_enriched, on=_ENRICHMENT_ID, how='left')\
+        .drop(_ENRICHMENT_ID, axis=1)
+
+    bq_client.delete_table(data_tablename, project=_WORKING_PROJECT, dataset=user_workspace)
+
+    return data_copy
+
+
+def __prepare_sql(enrichment_id, variables, enrichment_table, enrichment_geo_table, user_workspace,
```
Comment (oleurud) [inline cartoframes/data/enrichment/points_enrichment.py]: A detail: in order to set a method as private, we are using only one dash `_prepare_sql`. I don't know if we should use 2 or 1 but we should use always the same

File: cartoframes/data/clients/bigquery_client.py (hunk 1 above)
Comment (alejandrohall) [reply cartoframes/data/clients/bigquery_client.py]: I will add TTL when uploading table and avoid deleting at the end of the process

File: cartoframes/data/enrichment/points_enrichment.py (hunk 4 above)
Comment (alejandrohall) [reply cartoframes/data/enrichment/points_enrichment.py]: We should be using two underscores, using just one is a convention to treat that method as a private, but you are relying on the developer. Using two underscores actually mangle the method and hide it. https://dbader.org/blog/meaning-of-underscores-in-python

File: cartoframes/data/enrichment/points_enrichment.py (hunk 2 above)
Comment (alejandrohall) [reply cartoframes/data/enrichment/points_enrichment.py]: Good catch!

File: cartoframes/data/clients/bigquery_client.py (hunk 1 above)
Comment (alrocar) [reply cartoframes/data/clients/bigquery_client.py]: No needed, we want to deal with this in a separate issue. Don't worry about that by now.

**Comment by alrocar** [comment]
Thanks @oleurud 

Yep, let's focus on small and frequent PRs tackling one thing at a time and tied to well defined issues. That way we can iterate faster.

@alejandrohall Great job BTW!! 🚀 💪

**Pull request merged**